### ISSUE TITLE
MutualLift for SBNSupports

### ISSUE THREAD

**Issue by matsen** [phylovi/libsbn#269]
* Once #268 is merged, we will have nice `SBNSupport` objects describing the subsplit support in the rooted and unrooted cases.
* Once #267 is done we will be able to lift bitsets.

The next step towards mutualization is to be able to take two `SBNSupports`, of compatible rooted/unrooted status, and lift each of them to the union of the taxon sets. Note that the taxon sets are built into the SBNSupport classes as string vectors. I propose this be a static function that takes the two SBNSupports and makes a new one.

So `SBNSupport::MutualLift(SBNSupport support1, SBNSupport support2)`, returning `lifted_support`, could look like

* Identify the taxa that appear in `support2` that don't appear in `support1`. Make `lifted_support` have taxa equal to `support1`'s taxa with these taxa appended to the end.
* Lift all of the bitsets accordingly. That is, if we have a bitset that is a key, lift it and have it map to the same value. Correspondingly for values.

_This is not going to mutualize things_. 

Eventually we will need to go back and insert entries into the maps for mutualization. But we have to do this first, so it seems worth it to have a separate issue/PR.

Notes:

* This version of `MutualLift` is not commutative. We could make it so by sorting the taxa. Would that be better?
* There will be some annoyingness having to do with the fact (I think it's a fact!) that the desired behavior is the same for both unrooted and rooted versions, but we want this to appear in both the rooted and unrooted subclasses of `SBNSupport`. So my proposal is to define a constructor for `RootedSBNSupport` that takes an `SBNSupport` by move and moves the new indexer, etc, into place. Then we can easily define a version for `RootedSBNSupport` that calls the version in the superclass and then makes a `RootedSBNSupport` out of it.

**Comment by matsen** [comment]
Comments from @mdkarcher :

* he suggests "Expand" vs "Lift" : 👍 
* we could make taxon mapping explicit-- let's factor this out by having a version that takes an explicit mapping, and another that makes that mapping for you
